Update docs for uenv v10 (#395)
eth-cscs/cscs-docs: docs/software/uenv/management.md

### Creating and using repositories

A repo will automatically be created in your [Scratch path][ref-storage-scratch] attached to the cluster you are on when you first use uenv.
This _default repo_ is used by all calls to uenv, unless it is overridden in a configuration file or using the `--repo` CLI flag, both documented below.

!!! question "Where is my repo?"
    The Scratch filesystem used depends on the cluster:

    | cluster | repo path |
    | ------- | --------- |
    | Eiger, Daint | `/ritom/scratch/cscs/$USER/.uenv-images` |
    | Santis       | `/capstor/scratch/cscs/$USER/.uenv-images` |
    | Clariden     | `/iopsstor/scratch/cscs/$USER/.uenv-images` |
    | Others       | `$SCRATCH/.uenv-images` (falls back to `$HOME/.uenv/repo`) |

[](){#ref-uenv-repo-multiple}
### Multiple repositories

> <span class="v-badge">uenv v10</span>

Reviewer: ```suggestion
<span class="v-badge">uenv v10.0</span>
```

uenv supports multiple named repositories, which is useful for sharing images within a team or project.

> Repos are configured in the [uenv configuration file][ref-uenv-configure-options-repos] using the `[[repositories]]` array:

Reviewer: ```suggestion
Repositories are configured in the [uenv configuration file][ref-uenv-configure-options-repos] using the `[[repositories]]` array:
```

```toml title="~/.config/uenv/config.toml"
[[repositories]]
name = "test"
path = "/capstor/scratch/cscs/username/test-repo"

[[repositories]]
name = "team"
path = "/capstor/store/cscs/userlab/sm42/uenv-images"
priority = 30
```

> With multiple repos configured, `uenv image ls` searches all of them and label each result with its source repo.

Reviewer: ```suggestion
With multiple repositories configured, `uenv image ls` searches all of them and label each result with its source repo.
```

> ??? example "uenv image ls output when multiple repos are configured"

Reviewer: ```suggestion
??? example "uenv image ls output when multiple repositories are configured"
```

    ```console
    $ uenv image ls
    repo default
      uenv                        arch   system  id                size(MB)  date
      cp2k/2025.1:v1              gh200  daint   2a56f1df31a4c196   2,693    2025-03-01
      prgenv-gnu/26.2:v2          gh200  daint   c71fd8d678a6c217   5,191    2026-02-10

    repo test
      uenv                        arch   system  id                size(MB)  date
      prgenv-gnu/25.6:v2          gh200  daint   b81fd6ba25e88782   4,191    2025-04-10

    repo team
      uenv                        arch   system  id                size(MB)  date
      gromacs/2024:v1             gh200  daint   b58e6406810279d5   3,658    2025-02-12
    ```

> !!! info "where is the `default` repo defined?"

Reviewer: ```suggestion
!!! info "where is the `default` repository defined?"
```

>     The default repo is determined by uenv according to which cluster it is running on, which file systems are attached, and whether those file systems already contain a repository.

Reviewer: ```suggestion
    The default repository is determined by uenv according to which cluster it is running on, which file systems are attached, and whether those file systems already contain a repository.
```

>     You can check where it has been set on the current system using the `uenv repo status` command to list all repos.

Reviewer: ```suggestion
    You can check where it has been set on the current system using the `uenv repo status` command to list all repositories.
```

[](){#ref-uenv-repo-priority}

> ### Repo priority

Reviewer: ```suggestion
### Repository priority
```

> The uenv that is selected by the `uenv start`, `uenv run` and the Slurm `--uenv` flag is affected by repo priority.

Reviewer: ```suggestion
The uenv that is selected by the `uenv start`, `uenv run` command and the Slurm `--uenv` flag is affected by repository priority.
```

> Uenv will search through each repo in priority order, and take the first match that it finds.

Reviewer: ```suggestion
Uenv will search through each repository in priority order, and take the first match that it finds.
```

> That is, if it finds a match in the highest priority repo, it will use that without searching for matches in the other repos.

Reviewer: technically, it's the lowest priority, but highest precedence. I know, low numbers for high precedence make it hard to write the sentence correctly.

Reviewer: Also
```suggestion
That is, if it finds a match in the highest priority repo, it will use that without searching for matches in the other repositories.
```

> Commands that modify the repo contents (`uenv image pull`, `uenv image add` and `uenv image rm`) always write to the repo with the highest priority.

Reviewer: ```suggestion
Commands that modify the repository contents (`uenv image pull`, `uenv image add` and `uenv image rm`) always write to the repository with the highest priority.
```

> To write to a different repo, use the `--repo` flag with the repo name:

Reviewer: ```suggestion
To write to a different repository, use the `--repo` flag with the repository name:
```

> ??? example "pull an image into the team repo"

Reviewer: ```suggestion
??? example "pull an image into the team repository"
```

    ```
    uenv --repo=team image pull cp2k/2025.1:v1
    ```
See the documentation for the [`--repo`][ref-uenv-repo-flag] flag below for more information about how to customize the search order.

> !!! info "what priority does a repo have?"

Reviewer: ```suggestion
!!! info "what priority does a repository have?"
```

>     If no priority is specified when creating a repo in a configuration file, it is given priority 10.

Reviewer: ```suggestion
    If no priority is specified when creating a repository in a configuration file, it is given priority 10.
```

>     The `default` repo is given priority 9, so it will have higher precedence.

Reviewer: ```suggestion
    The `default` repository is given priority 9, so it will have higher precedence.
```

[](){#ref-uenv-repo-flag}
### The `--repo` flag

> The `--repo` flag selects which repos to use for a command.

Reviewer: ```suggestion
The `--repo` flag selects which repositories to use for a command.
```

It goes between `uenv` and the subcommand and can be used with all uenv commands.

> It accepts a comma-separated list of repos, where each repo is one of a repo name (from the config file), an absolute path, or a `name=path` pair.

Reviewer: ```suggestion
It accepts a comma-separated list of repositories, where each repository is one of a repository name (from the config file), an absolute path, or a `name=path` pair.
```

> !!! example "selecting repos with `--repo`"

Reviewer: ```suggestion
!!! example "selecting repositories with `--repo`"
```

    ```console
    # by name (defined in config)
    $ uenv --repo=team image ls

    # by path (no config needed)
    $ uenv --repo=/capstor/store/cscs/userlab/sm42/uenv-images image ls

    # explicit name=path (ad-hoc, overrides config entirely)
    $ uenv --repo=team=/capstor/store/cscs/userlab/sm42/uenv-images image ls

>     # multiple repos in order of priority

Reviewer: ```suggestion
    # multiple repositories in order of priority
```

    $ uenv --repo=test,team image ls
    ```

> The order of repos in a `--repo` list is important: it defines the order of precedence of the repos.

Reviewer: ```suggestion
The order of repositories in a `--repo` list is important: it defines the order of precedence of the repositories.
```

> This means that operations that select a uenv, for example `uenv start`, will search through the repos in the order provided and select 

Reviewer: ```suggestion
This means that operations that select a uenv, for example `uenv start`, will search through the repositories in the order provided and select 
```

!!! tip "use `--repo` to override precedence"

>     Repos in a `--repo` list are searched in the order that they are written in the list.

Reviewer: ```suggestion
    Repositories in a `--repo` list are searched in the order that they are written in the list.
```

>     This is very useful if you want to change the default search order, for example the `default` repo typically has higher precedence than other repos.

Reviewer: ```suggestion
    This is very useful if you want to change the default search order, for example the `default` repository typically has higher precedence than other repositories.
```

>     The following example will changes the search order:

Reviewer: ```suggestion
    The following example changes the search order:
```

    ```bash
    uenv --repo=team,default start cp2k/2025.1
    ```

>     If there is a unique image matching `cp2k/2025.1` in the `team` repo, the `default` repo will not be checked.

Reviewer: ```suggestion
    If there is a unique image matching `cp2k/2025.1` in the `team` repository, the `default` repository will not be checked.
```
